src/common/k8s-api/kube-api.ts

```diff
@@ -248,7 +245,7 @@ export function forRemoteCluster<
   }
 
   if (config.agent) {
-    reqInit.agent = config.agent;
+    reqInit.agent = config.agent.constructor(agentOptions);
```

Comment:
No `new` is needed?

Reply:
No.. but this doesn't really work.. so removed that line.